ignite/pkg/cosmosbuf/buf.go

```diff
@@ -67,6 +69,45 @@ func (c Command) String() string {
 	return string(c)
 }
 
+// Export runs the buf Export command for the files in the proto directory.
+func (b Buf) Export(
+	ctx context.Context,
+	protoDir,
+	output string,
+) (err error) {
+	flags := map[string]string{
+		flagOutput: output,
+	}
+	g, ctx := errgroup.WithContext(ctx)
+
+	if strings.Contains(protoDir, cosmosver.CosmosModulePath) {
+		if b.sdkCache == "" {
+			b.sdkCache, err = PrepareSDK(protoDir)
+			if err != nil {
+				return err
+			}
+		}
+		dirs := strings.Split(protoDir, "/proto")
+		if len(dirs) < 2 {
+			return fmt.Errorf("invalid cosmos sdk mod path: %s", dirs)
+		}
+		protoDir = filepath.Join(b.sdkCache, dirs[1])
+	}
```

Comment:
Maybe:

```suggestion
	if strings.HasPrefix(protoDir, cosmosver.CosmosModulePath) {
		if b.sdkCache == "" {
			b.sdkCache, err = PrepareSDK(protoDir)
			if err != nil {
				return err
			}
		}

		path := strings.TrimPrefix(protoDir, cosmosver.CosmosModulePath+"/proto")
		if path == "" {
			return fmt.Errorf("invalid cosmos SDK mod path: %s", protoDir)
		}
		protoDir = filepath.Join(b.sdkCache, path)
	}

```

Reply:
I just saw other methods implement something like what you implemented. I added this suggestion thinking that this part could be easier to understand by using `HasPrefix` and `TrimPrefix`

Reply:
I copied that implementation from the Generate one. Ideally shoudl be refactored to a separate function and implemented like you suggest.

Planning to do that along with handling the temp folders the right way